src/include/sof/lib/dma.h

```diff
@@ -541,8 +541,8 @@ int dma_copy_from_host_nowait(struct dma_copy *dc,
 			      int32_t size);
 
 /* DMA copy data from DSP to host */
-int dma_copy_to_host_nowait(struct dma_copy *dc, struct dma_sg_config *host_sg,
-	int32_t host_offset, void *local_ptr, int32_t size);
+int dma_copy_to_host(struct dma_copy *dc, struct dma_sg_config *host_sg,
+		     int32_t host_offset, void *local_ptr, int32_t size);
```

Comment:
Just to confirm, are we changing how any clients call this code i.e. do clients now block ? (or were they already blocking as the function name was not aligned to real logic)

Reply:
Yes, but there is only two clients and they are both covered in the commit. The first is the dtrace code and then the probe.c is the "collateral damage". When reading the probe.c I could not find any reason why the dma there should not block. In fact I think that code could also potentially suffer from a similar race condition.

Reply:
I can, of course, but since there are really no real users for the *_nowait version, I am not sure I should.